### PR TITLE
Ai seeder

### PR DESCRIPTION
**Pull request by mas-iota**

### CODE REVIEW

File: src/control/content/containers/Content.js [hunk 1]
```diff
@@ -329,24 +330,36 @@ class Content extends Component {
     );
   };
 
+  saveSampleAiData = (state, type) => {
```
Comment (mas-iota) [inline src/control/content/containers/Content.js]: reset? true | false
Function name -> onSaveAIData

Investigate if we can import the state parameters pass folders/images from that state, {images, folders}

File: src/control/content/containers/Content.js [hunk 2]
```diff
@@ -329,24 +330,36 @@ class Content extends Component {
...
-
       let originalState = JSON.parse(JSON.stringify({ ...result.data }))
 
-      if (images && folders) {
-        this.setState(() => ({ images, folders, originalState }));
-      }
+      if ((images && images.length) || (folders && folders.length) ) {
+        this.setState(() => ({ images, folders, originalState, loaded: true }));
+      } 
+      else
```
Comment (mas-iota) [inline src/control/content/containers/Content.js]: guard the else with brackets

File: src/control/content/containers/Home.js [hunk 3]
```diff
@@ -1,33 +1,254 @@
...
+  if (loaded && showEmptyState) {
+    const generateRandomNumber = () => {
+      const min = 1000000000000; // Smallest 13-digit number
+      const max = 9999999999999; // Largest 13-digit number
+      return Math.floor(Math.random() * (max - min + 1)) + min;
+    };
+
+    /** UTILS */
+    const checkImageUrl = (url) => {
+      const optimisedURL = url.replace('1080x720', '300x300');
```
Comment (mas-iota) [inline src/control/content/containers/Home.js]: Smaller is better

File: src/control/content/containers/Home.js [hunk 4]
```diff
@@ -1,33 +1,254 @@
...
+
+    /** UTILS */
+    const checkImageUrl = (url) => {
+      const optimisedURL = url.replace('1080x720', '300x300'); 
+      return new Promise((resolve, reject) => {
+        if (url.includes("http")){
+          const xhr = new XMLHttpRequest();
+          xhr.open("GET", optimisedURL);
+          xhr.onerror = (error) => {
+            console.error('provided URL is not a valid image', error);
```
Comment (mas-iota) [inline src/control/content/containers/Home.js]: Not necessary, console.warn

File: src/control/content/containers/Home.js [hunk 5]
```diff
@@ -1,33 +1,254 @@
...
+  // init state seeder
+  if (loaded && showEmptyState) {
+    const generateRandomNumber = () => {
+      const min = 1000000000000; // Smallest 13-digit number
+      const max = 9999999999999; // Largest 13-digit number
+      return Math.floor(Math.random() * (max - min + 1)) + min;
+    };
+
+    /** UTILS */
+    const checkImageUrl = (url) => {
```
Comment (mas-iota) [inline src/control/content/containers/Home.js]: elimanateNotFoundImages()

File: src/control/content/containers/Home.js [hunk 6]
```diff
@@ -1,33 +1,254 @@
...
+                message: "Bad AI request, please try changing your request.",
+                type: "danger",
+              });
+            }
+
+            Promise.all(promises).then(items => {
+              let processedImages = items.map(
+                ({ naturalWidth, naturalHeight, originalSrc }) => {
+                  return  new Img({
+                      src: originalSrc,
```
Comment (mas-iota) [inline src/control/content/containers/Home.js]: use the fetched URLs

File: src/control/content/containers/Home.js [hunk 7]
```diff
@@ -1,33 +1,254 @@
...
+            );
+            if (!promises.length){
+              seeder.requestResult.complete();
+              return buildfire.dialog.toast({
+                message: "Bad AI request, please try changing your request.",
+                type: "danger",
+              });
+            }
+
+            Promise.all(promises).then(items => {
```
Comment (mas-iota) [inline src/control/content/containers/Home.js]: Promise.allSettled()

File: src/control/content/containers/Home.js [hunk 8]
```diff
@@ -1,33 +1,254 @@
...
+      },
+      importOptions: {
+        jsonTemplate: {
+          folders: [{ name: '', images: [{ url: '' }] }]
+        },
+        sampleCSV: 'Weather: https://source.unsplash.com/featured/?sunny\n'
+          + 'Weather:https://source.unsplash.com/featured/?rainy\n'
+          + 'City:https://source.unsplash.com/featured/?street',
+        systemMessage: 'Each row contains a folder and image URL that should belong to. if nothing provided or no URLs provided then return null',
+        maxRecords: 10,
```
Comment (mas-iota) [inline src/control/content/containers/Home.js]: make it 25

**Review by mas-iota — COMMENTED** [review]
Request Changes

File: src/control/content/containers/Home.js (hunk 8 above)
Comment (abdalladimes) [reply src/control/content/containers/Home.js]: I've made it 5 after testing, it's the most stable choice

**Pull request merged**